docs/src/main/asciidoc/rest-json-guide.adoc

```diff
@@ -321,7 +321,7 @@ import javax.ws.rs.core.Response;
 @Consumes(MediaType.APPLICATION_JSON)
 public class LegumeResource {
 
-    private Set<Legume> legumes = Collections.newSetFromMap(Collections.synchronizedMap(new LinkedHashMap<>()));
+    private Set<Legume> legumes = Collections.synchronizedSet(new LinkedHashSet<>());
```

Comment:
Any reason to change this?

Reply:
No one is constructing synchronized set this way :) If you want concurrent access then the the correct way is:

```java
Collections.newSetFromMap(new ConcurrentHashMap<>());
```

And if you want synchronized access then it should be done like this:

```java
Collections.synchronizedSet(new LinkedHashSet<>());
```

Having:

```java
Collections.newSetFromMap(Collections.synchronizedMap(new LinkedHashMap<>()));
```

Is an exaggerated use of `Collections` API.

Reply:
Agreed that I was not on my best day when I wrote this :).